integration/kube_integration_test.go

```diff
@@ -593,6 +595,196 @@ func testKubePortForward(t *testing.T, suite *KubeSuite) {
+				// 	pod, err := suite.CoreV1().Pods(testNamespace).Get(context.Background(), testPod, metav1.GetOptions{})
+				// 	if err != nil {
+				// 		return false
+				// 	}
+				// 	if pod.Status.Phase != v1.PodRunning {
+				// 		return false
+				// 	}
+				// 	if len(pod.Status.ContainerStatuses) == 0 {
+				// 		return false
+				// 	}
+				// 	for _, containerStatus := range pod.Status.ContainerStatuses {
+				// 		if containerStatus.Name == nginx {
+				// 			return containerStatus.Ready
+				// 		}
+				// 	}
+				// 	return false
+				// }, 60*time.Second, time.Millisecond*500)
+
+				t.Cleanup(func() {
+					// Test pod would have been deleted if test is successful.
+					// Setup next test with test pod as needed.
+					pod := newPod(testNamespace, testPod)
+					_, err = suite.CoreV1().Pods(testNamespace).Create(context.Background(), pod, metav1.CreateOptions{})
+					if err != nil {
+						require.True(t, kubeerrors.IsAlreadyExists(err), "Failed to create test pod: %v", err)
+					}
+					// Wait for pod to be running.
+					require.Eventually(t, func() bool {
+						rsp, err := suite.CoreV1().Pods(testNamespace).Get(context.Background(), testPod, metav1.GetOptions{})
+						if err != nil {
+							return false
+						}
+						return rsp.Status.Phase == v1.PodRunning
+					}, 60*time.Second, time.Millisecond*500)
+				})
+
+				// Forward local port to container port.
+				listener, err := net.Listen("tcp", "localhost:0")
+				require.NoError(t, err)
+				t.Cleanup(func() {
+					require.NoError(t, listener.Close())
+				})
+				localPort := listener.Addr().(*net.TCPAddr).Port
+				forwarder, err := tt.builder(proxyClientConfig, kubePortForwardArgs{
+					ports:        []string{fmt.Sprintf("%v:80", localPort)},
+					podName:      testPod,
+					podNamespace: testNamespace,
+				})
+				require.NoError(t, err)
+
+				forwarderCh := make(chan error, 1)
+				go func() {
+					err := forwarder.ForwardPorts()
+					forwarderCh <- err
+					if err != nil {
+						t.Logf("Port forwarding error: %v", err)
+					}
+				}()
+
+				// Wait for port-forwarding to be ready.
+				start := time.Now()
+				select {
+				case <-time.After(5 * time.Second):
+					t.Fatalf("Timeout waiting for port forwarding after %v", time.Since(start))
+				case err := <-forwarderCh:
+					t.Fatalf("Port forwarding failed immediately: %v", err)
+				case <-forwarder.readyC:
+					t.Logf("Port forwarding ready after %v", time.Since(start))
+				}
+
+				// Validate that port-forwarding is working.
+				resp, err := http.Get(fmt.Sprintf("http://localhost:%v", localPort))
+				require.NoError(t, err)
+				require.Equal(t, http.StatusOK, resp.StatusCode)
+				require.NoError(t, resp.Body.Close())
+
+				// Delete the pod.
+				err = suite.CoreV1().Pods(testNamespace).Delete(context.Background(), testPod, metav1.DeleteOptions{})
+				require.NoError(t, err)
+
+				impersonatingForwarder, err := tt.builder(impersonatingProxyClientConfig, kubePortForwardArgs{
```

Comment:
you can discard this section and ensure `forwarderCh` exits after deleting the pod